tv_apps/browser/js/tooltip.js

```diff
@@ -0,0 +1,186 @@
+/* global _ */
+/* global Toolbar */
+
+'use strict';
+
+/**
+ * Browser App Tooltip.
+ * @namespace Tooltip
+ */
+var Tooltip = {
+  tooltipBlock: null,
+  showTimeoutID: null,
+  hideTimeoutID: null,
+  hideTimeoutData: 100,
+  overObj: null,
+  moveObj: null,
+  showTarget: null,
+
+  /**
+   * initialize
+   */
+  init: function tooltip_init() {
+    this.tooltipBlock = document.getElementById('tooltip-block');
+
+    var elements = [
+      {target: Toolbar.sidebarButtonBlock},
+      {target: Toolbar.backButtonBlock},
+      {target: Toolbar.forwardButtonBlock},
+      {target: Toolbar.addressButton},
+      {target: Toolbar.searchInput},
+      {target: Toolbar.bookmarkButtonBlock},
+      {target: Toolbar.showBookmarksButtonBlock},
+      {target: Toolbar.homeButtonBlock},
+      {target: Toolbar.zoomButtonBlock},
+      {target: Toolbar.tabsButtonBlock},
+      {target: Toolbar.newTabButtonBlock},
+      {target: Toolbar.menuButtonBlock},
+      {target: Toolbar.modeButtonBlock},
+    ];
+
+    this.overObj = this.mouseoverEvent.bind(this);
+    this.moveObj = this.mousemoveEvent.bind(this);
+    // toolbar icon
+    elements.forEach(function addEvent(elm) {
+      var target = elm.target;
+      // add event
+      target.addEventListener('mouseover', this.overObj);
+      target.addEventListener('mouseout', this.mouseoutEvent.bind(this));
+      target.addEventListener('mousedown', this.clickEvent.bind(this));
+      target.addEventListener('mouseup', this.clickEvent.bind(this));
+    }, this);
+  },
+
+  /**
+   * mousemove event
+   */
+  mousemoveEvent: function tooltip_mousemoveEvent(ev) {
+    if(this.isShow()) {
+      // hide the tooltip in mouse movement at tooltip display.
+      this.stopMousemoveEvent(ev.target);
+      this.hideTimeoutID = setTimeout(function(){
+        Tooltip.hide();
+      }, this.hideTimeoutData);
+      return;
+    }
+    if(this.showTimeoutID) {
+      clearTimeout(this.showTimeoutID);
+    }
+
+    var bounds = ev.currentTarget.getBoundingClientRect();
+    var posY = Math.ceil(bounds.bottom);
+    var top = Math.ceil(bounds.top);
+    var tips = null;
+    if(ev.target.dataset.tips != null && ev.target.dataset.tips != undefined) {
+      if(ev.target == Toolbar.searchInput) {
+        tips = ev.target.dataset.tips;
+      } else {
+        tips = _(ev.target.dataset.tips);
+      }
+    }
```

Comment:
this should be:
`
if(ev.target == Toolbar.searchInput) {
  tips = {raw: ev.target.dataset.tips};
} else {
  tips = ev.target.dataset.tips;
}
`

and then in Tooltip.displaySetting it should be sth like:

`
if (disp.text.raw) {
  element.textContent = disp.text.raw;
} else {
  element.setAttribute('data-l10n-id', disp.text);
}
`

See: https://developer.mozilla.org/en-US/Firefox_OS/Developing_Gaia/localization_code_best_practices#Writing_APIs_that_operate_on_L10nIDs